azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -414,7 +459,147 @@ private void setupKeyStore() throws ExecutorManagerException {
         throw new ExecutorManagerException("Failed to delete the cert file");
       }
     }
+  }
+
+  @VisibleForTesting
+  void start() {
+     this.containerContext.setAttribute(Constants.AZKABAN_CONTAINER_CONTEXT_KEY, this);
+  }
+
+  public void cancelFlow(final int execId, final String user)
+    throws ExecutorManagerException {
+
+    logger.info("Cancel Flow called");
+    if (this.flowRunner == null) {
+      throw new ExecutorManagerException("Flow has not launched yet.");
+    }
+
+    if (Status.isStatusFinished(this.flowRunner.getExecutableFlow().getStatus())) {
+      logger.warn("Found a finished execution in the list of running flows: " + execId);
+      throw new ExecutorManagerException("Execution is already finished.");
+    }
+
+    this.flowRunner.kill(user);
+  }
+
+  /**
+   * Return accumulated flow logs with the specified length from the flow container starting from the given byte offset.
+   * @param execId
+   * @param startByte
+   * @param length
+   * @return
+   * @throws ExecutorManagerException
+   */
+  public LogData readFlowLogs(final int execId, final int startByte, final int length)
+    throws ExecutorManagerException {
+    logger.info("readFlowLogs called");
+    if (this.flowRunner == null) {
+      throw new ExecutorManagerException("The flow has not launched yet!");
+    }
+
+    final File dir = flowRunner.getExecutionDir();
+    if (dir != null && dir.exists()) {
```

Comment:
Can you restructure this to avoid indentation with negative case handled here instead of line 514. This way the try catch block is not inside any condition. Applies to readJobLogs and readJobMetadata too.
Something like,
if (dir == null || !dir.exists() ) {
  throw;
} 

try {}
catch{}